cpp/ql/src/semmle/code/cpp/Function.qll

```diff
@@ -393,18 +393,23 @@ class Function extends Declaration, ControlFlowNode, AccessHolder, @function {
 
   /** Gets a function that overloads this one. */
   Function getAnOverload() {
-    result.getName() = getName() and
-    result.getNamespace() = getNamespace() and
-    result != this and
-    // If this function is declared in a class, only consider other
-    // functions from the same class. Conversely, if this function is not
-    // declared in a class, only consider other functions not declared in a
-    // class.
     (
-      if exists(getDeclaringType())
-      then result.getDeclaringType() = getDeclaringType()
-      else not exists(result.getDeclaringType())
+      // If this function is declared in a class, only consider other
+      // functions from the same class.
+      exists(string name, Namespace namespace, Class declaringType |
+        candGetAnOverloadMember(name, namespace, declaringType, this) and
+        candGetAnOverloadMember(name, namespace, declaringType, result)
```

Comment:
I believe there's no need for a `namespace` parameter on `candGetAnOverloadMember`. If two functions are in the same class, they must also be in the same namespace.

Reply:
Good point. Fixed in e4726db52200df99b5aa005ee044d3b251b5acc4.